data/block/blockV2.go

```diff
@@ -0,0 +1,617 @@
+//go:generate protoc -I=proto -I=$GOPATH/src -I=$GOPATH/src/github.com/ElrondNetwork/protobuf/protobuf  --gogoslick_out=. blockV2.proto
+package block
+
+import (
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/data"
+)
+
+// GetShardID returns the header shardID
+func (hv2 *HeaderV2) GetShardID() uint32 {
+	if hv2 == nil || hv2.Header == nil {
+		return 0
+	}
+
+	return hv2.Header.ShardID
+}
+
+// GetNonce returns the header nonce
+func (hv2 *HeaderV2) GetNonce() uint64 {
```

Comment:
I think all getters could have been written like:
```GO
func (hv2 *HeaderV2) GetNonce() uint64 {
	if hv2 == nil { 
		return 0
	}
	
	return hv2.Header.GetNonce()
}
```
as an alternative to proposed rework, maybe extract the condition `hv2 == nil || hv2.Header == nil` in a function?